src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -49,6 +50,12 @@ protected override void InitializeCompareExchangeStorage()
         CompareExchangeStorage.Initialize(ShardedDatabaseName);
     }
 
+    protected override void InitializeAndStartPeriodicDocumentsMigrator()
+    {
+        PeriodicDocumentsMigrator = new ShardedPeriodicDocumentsMigrator(this);
```

Comment:
Do we need to dispose this on shutdown?
If we change the time interval setting, would be propagated to this instance?

Reply:
I think the `Dispose()` in `BackgroundWorkBase` is enough.
It will propagate to the instance but only after restarting the DB :-)

Reply:
do we call the dispose?

Reply:
done